src/game/client/components/chat.cpp

```diff
@@ -196,7 +196,30 @@ void CChat::ClearChatBuffer()
 
 bool CChat::OnInput(IInput::CEvent Event)
 {
-	if(m_Mode == CHAT_NONE || Client()->State() != Client()->STATE_ONLINE)
+	if(Client()->State() != Client()->STATE_ONLINE)
+		return false;
+
+	// chat history scrolling
+	if(m_Show && Event.m_Flags&IInput::FLAG_PRESS && (Event.m_Key == KEY_PAGEUP || Event.m_Key == KEY_PAGEDOWN))
+	{
+		if(Event.m_Flags&IInput::FLAG_PRESS)
+		{
+			if(Event.m_Key == KEY_PAGEUP)
+			{
+				++m_BacklogPage;
+				if(m_BacklogPage >= MAX_CHAT_PAGES) // will be further capped during rendering
+					m_BacklogPage = MAX_CHAT_PAGES-1;
+			}
+			else if(Event.m_Key == KEY_PAGEDOWN)
+			{
+				--m_BacklogPage;
+				if(m_BacklogPage < 0)
+					m_BacklogPage = 0;
+			}
+		}
+		return true;
+	}
+	if(m_Mode == CHAT_NONE)
```

Comment:
This check should probably be up, so you don't trigger it when the chat is closed. Guess ingame binds for PageUp/Down don't work either?